### PR TITLE
Rename some movement actions and functions

### PR DESCRIPTION
**Pull request by cbt6**
Mostly done so that movements can ported more easily to emerald.

Names were taken from pokeemerald where possible.

### CODE REVIEW

File: include/constants/event_object_movement.h [hunk 1]
```diff
@@ -235,33 +235,33 @@
...
-#define MOVEMENT_ACTION_ACRO_END_WHEELIE_MOVE_UP        0x95
-#define MOVEMENT_ACTION_ACRO_END_WHEELIE_MOVE_LEFT      0x96
-#define MOVEMENT_ACTION_ACRO_END_WHEELIE_MOVE_RIGHT     0x97
 
-// 98-A3 are unknown
+#define MOVEMENT_ACTION_SPIN_DOWN  0x94
+#define MOVEMENT_ACTION_SPIN_UP    0x95
+#define MOVEMENT_ACTION_SPIN_LEFT  0x96
+#define MOVEMENT_ACTION_SPIN_RIGHT 0x97
+
```
Comment (GriffinRichards) [inline include/constants/event_object_movement.h:243]: The spaces between adjacent groups (`93`/`94`, `97`/`98`, etc) aren't necessary, the groups are clear from the names.

File: src/event_object_movement.c [hunk 2]
```diff
@@ -2591,23 +2590,22 @@ u16 GetObjectEventFlagByLocalIdAndMap(u8 localId, u8 mapNum, u8 mapGroup)
...
 }
 
-u8 sub_808F0BC(u8 objectEventId)
+u8 Unused_GetObjectTrainerTypeByObjectEventId(u8 objectEventId)
 {
     return gObjectEvents[objectEventId].trainerType;
 }
 
-u8 sub_805FD08(u8 localId, u8 mapNum, u8 mapGroup)
+u8 Unused_GetObjectEventBerryTreeIdByLocalIdAndMap(u8 localId, u8 mapNum, u8 mapGroup)
```
Comment (GriffinRichards) [inline src/event_object_movement.c]: All these `Unused` can just be a comment rather than part of the name

File: asm/macros/movement.inc [hunk 3]
```diff
@@ -14,10 +14,10 @@
...
 	create_movement face_left_fast
 	create_movement face_right_fast
-	create_movement walk_slowest_down
-	create_movement walk_slowest_up
-	create_movement walk_slowest_left
-	create_movement walk_slowest_right
+	create_movement walk_slower_down
+	create_movement walk_slower_up
+	create_movement walk_slower_left
+	create_movement walk_slower_right
```
Comment (GriffinRichards) [inline asm/macros/movement.inc:20]: Should also create movement macros for the newly named movements (even though they're not used by vanilla scripts)

**Comment by cbt6** [comment]
Just realized movement action 0x9F was named "shake head" which shouldn't be the case since that only applies for the male character, any suggestions on what it could be named instead?
unknown unknown

**Comment by GriffinRichards** [comment]
Interesting! I also only tested this initially with the male player. This is the only case where their animations differ like this. Is it actually used anywhere?

**Comment by cbt6** [comment]
Not that I'm aware of.

**Comment by GriffinRichards** [comment]
Tough to say without the context of its intention. Perhaps when they decided against using it they simply left placeholders for the female sprite rather than drawing new images that would be unused. Or maybe it's some kind of test animation. The "obvious" combination name `MOVEMENT_ACTION_SHAKE_HEAD_OR_WALK_IN_PLACE` is a bit unwieldy.

**Comment by cbt6** [comment]
Think `MOVEMENT_ACTION_SHAKE_HEAD_OR_WALK_IN_PLACE` should be fine; if anything, there are already other movement actions such as `MOVEMENT_ACTION_ENABLE_JUMP_LANDING_GROUND_EFFECT` and `MOVEMENT_ACTION_JUMP_SPECIAL_WITH_EFFECT_DOWN` which have longer names.

**Pull request merged**